### ISSUE TITLE
[question] Buildinfo jenkins artifactory plugin throws errors

### ISSUE THREAD

**Issue by maitrey** [conan-io/conan#12683]
Dear Conan Folks,
While uploading the buildInfo throws exception:
11:10:27  Added for debugging purpose after command uploadorg.jfrog.hudson.pipeline.common.types.buildInfo.BuildInfo@XXXXXXX
[Pipeline] runConanCommand
[Pipeline] }
11:10:27  an exception which occurred:
11:10:27  	in field com.cloudbees.groovy.cps.impl.BlockScopeEnv.locals
11:10:27  	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@2488a5a1
11:10:27  	in field com.cloudbees.groovy.cps.impl.ProxyEnv.parent
11:10:27  	in object com.cloudbees.groovy.cps.impl.BlockScopeEnv@1988f43b
11:10:27  	in field com.cloudbees.groovy.cps.impl.CallEnv.caller
11:10:27  	in object com.cloudbees.groovy.cps.impl.FunctionCallEnv@2cb64106
11:10:27  	in field com.cloudbees.groovy.cps.Continuable.e
11:10:27  	in object org.jenkinsci.plugins.workflow.cps.SandboxContinuable@25651169
11:10:27  	in field org.jenkinsci.plugins.workflow.cps.CpsThread.program
11:10:27  	in object org.jenkinsci.plugins.workflow.cps.CpsThread@90fc2fb
11:10:27  	in field org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.threads
11:10:27  	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@2094b5c1
11:10:27  	in object org.jenkinsci.plugins.workflow.cps.CpsThreadGroup@2094b5c1
11:10:27  Caused: java.io.NotSerializableException: java.util.regex.Matcher
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:274)
11:10:27  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
11:10:27  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
11:10:27  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
11:10:27  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
11:10:27  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)
11:10:27  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)
11:10:27  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)
11:10:27  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)
11:10:27  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1080)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.BlockMarshaller.doWriteObject(BlockMarshaller.java:65)
11:10:27  	at org.jboss.marshalling.river.BlockMarshaller.writeObject(BlockMarshaller.java:56)
11:10:27  	at org.jboss.marshalling.MarshallerObjectOutputStream.writeObjectOverride(MarshallerObjectOutputStream.java:50)
11:10:27  	at org.jboss.marshalling.river.RiverObjectOutputStream.writeObjectOverride(RiverObjectOutputStream.java:179)
11:10:27  	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:341)
11:10:27  	at java.base/java.util.HashMap.internalWriteEntries(HashMap.java:1858)
11:10:27  	at java.base/java.util.HashMap.writeObject(HashMap.java:1412)
11:10:27  	at org.jboss.marshalling.reflect.JDKSpecific$SerMethods.callWriteObject(JDKSpecific.java:89)
11:10:27  	at org.jboss.marshalling.reflect.SerializableClass.callWriteObject(SerializableClass.java:199)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1089)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteFields(RiverMarshaller.java:1143)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteSerializableObject(RiverMarshaller.java:1101)
11:10:27  	at org.jboss.marshalling.river.RiverMarshaller.doWriteObject(RiverMarshaller.java:268)
11:10:27  	at org.jboss.marshalling.AbstractObjectOutput.writeObject(AbstractObjectOutput.java:58)
11:10:27  	at org.jboss.marshalling.AbstractMarshaller.writeObject(AbstractMarshaller.java:116)
11:10:27  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.lambda$writeObject$1(RiverWriter.java:144)
11:10:27  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.GroovySandbox.runInSandbox(GroovySandbox.java:331)
11:10:27  	at org.jenkinsci.plugins.workflow.support.pickles.serialization.RiverWriter.writeObject(RiverWriter.java:143)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:577)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgram(CpsThreadGroup.java:554)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.saveProgramIfPossible(CpsThreadGroup.java:537)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:461)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:330)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:294)
11:10:27  	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
11:10:27  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
11:10:27  	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)
11:10:27  	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:30)
11:10:27  	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:70)
11:10:27  	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
11:10:27  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
11:10:27  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
11:10:27  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
11:10:27  	at java.base/java.lang.Thread.run(Thread.java:829)
Code is as in the example: https://github.com/jfrog/project-examples/blob/master/jenkins-examples/pipeline-examples/scripted-examples/conan-example/Jenkinsfile 
I have checked that the build pattern is provided the same as in the pipeline. User has access to
Upload itself via command line is working(conan upload Test/1.0.5@staging/devel --all -r d15dec84-g77e-4577-8ae6-7a190eda2e2e --confirm). But i cannot understand the exception raised .
Any help/hints is greatly appreciated.
Thanks!

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @maitrey 

This seems an issue to be reported in https://github.com/jfrog/jenkins-artifactory-plugin, as we don't own this plugin.

It is true, that the BuildInfo capabilities of Conan haven't been updated in a while, and we will need to do that, but this is not a priority until 2.0 is out. I'd probably try to wait until 2.0 updates this.

**Comment by maitrey** [comment]
Hi @memsharded , Somehow the issue seems to come from the fact that I have two remotes set : One for PR/branch builds and the other for tag builds. Even if I push the upload to branch repo even then it doesnot work. Definitely, I am missing something.
In the pipeline job : I have set CONAN_USER_HOME as the current job path. 
Before starting any conan commands:
bat(script: 'conan config install https://github.com/conanconfig.git')
bat(script: 'conan user %ARTIFACTORY_CREDENTIALS_USR% -p  %ARTIFACTORY_CREDENTIALS_PSW% -r branch-remote')
bat(script: 'conan user %ARTIFACTORY_CREDENTIALS_USR% -p  %ARTIFACTORY_CREDENTIALS_PSW% -r release-remote')  

Do we need to set any other parameter if I download packages from another repo and upload to another repo?

**Comment by memsharded** [comment]
I am not sure how that configuration with remotes is related to the buildinfo posted above, sorry.
It is definitely possible to download packages from one repo and upload them to other repo.

**Comment by maitrey** [comment]
Hmm... not sure then what is the issue :, raised a question here:  https://github.com/jfrog/jenkins-artifactory-plugin/issues/724 .
Please close this one if nothing more needs to be added. Many Thanks!

**Comment by maitrey** [comment]
Hello @memsharded  , Can we please close the issue as nothing can be done in this case in this forum?

**Comment by memsharded** [comment]
Ok, closing this at the moment. At some point after Conan 2.0 is final, we will resume work on Artifactory build-info, on the plugins, etc, but at the moment those things would need to wait a bit.

**Comment by maitrey** [comment]
Thanks for closing the issue. I also fixed the issue which had nothing to do with buildinfo:
11:10:27 Caused: java.io.NotSerializableException: java.util.regex.Matcher
It was because I am using matcher to check if tag fits to semantic version which I use as the version of the package.
matcher is not serializable . So, one needs to specify explicitly so that jenkins pipeline engine can handle this natively.
